### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by maciej-sypetkowski**
## Changelogs

- Add `eval` resolver for OmegaConf for more scriptable configuration
- Fix pe_encoders that are neither `feat` nor `edge_feat`
- Add `out_dim_edges` for gnn network (previously it was forced to be the same as last `hidden_dims_edges`)
- Fix multiaggregation for mpnn when passing multiple aggregation methods
- Added a few non-ipu losses that had corresponding ipu function.

---

_Checklist:_

- [ ] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [ ] _Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate)._
- [ ] _Update the API documentation is a new function is added, or an existing one is deleted._
- [ ] _Write concise and explanatory changelogs above._
- [ ] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

_discussion related to that PR_

### CODE REVIEW

File: graphium/utils/spaces.py [hunk 1]
```diff
@@ -105,7 +108,7 @@
     "msle": TorchMetrics.mean_squared_log_error,
     "pearsonr": TorchMetrics.pearson_corrcoef,
     "spearmanr": TorchMetrics.spearman_corrcoef,
-    "r2": TorchMetrics.r2_score,
+    "r2_score": TorchMetrics.r2_score,
```
Comment (DomInvivo) [inline graphium/utils/spaces.py:111]: Please do a ctrl+F to change all `r2` in yaml files to `r2_score`

**Review by DomInvivo — APPROVED** [review]
Approved. Minor comment regarding `r2` -> `r2_score`

**Comment by DomInvivo** [comment]
To fix the linting, please run `black PATH_TO_REPO`,
Also, don't merge before the tests are all passing

File: graphium/utils/spaces.py (hunk 1 above)
Comment (maciej-sypetkowski) [reply graphium/utils/spaces.py:111]: interestingly `r2_score` was already used in many places

**Pull request merged**